specification/cdn/resource-manager/Microsoft.Cdn/2017-04-02/cdn.json

```diff
@@ -0,0 +1,2730 @@
+          "items": {
+            "$ref": "#/definitions/DeepCreatedOrigin"
+          }
+        },
+        "resourceState": {
+          "description": "Resource status of the endpoint.",
+          "readOnly": true,
+          "enum": [
+            "Creating",
+            "Deleting",
+            "Running",
+            "Starting",
+            "Stopped",
+            "Stopping"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "EndpointResourceState",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "description": "Provisioning status of the endpoint.",
+          "type": "string"
+        }
+      }
+    },
+    "EndpointListResult": {
+      "description": "Result of the request to list endpoints. It contains a list of endpoint objects and a URL link to get the the next set of results.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Endpoint"
+          },
+          "description": "List of CDN endpoints within a profile"
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to get the next set of endpoint objects if there is any."
+        }
+      }
+    },
+    "EndpointUpdateParameters": {
+      "type": "object",
+      "description": "Properties required to create or update an endpoint.",
+      "properties": {
+        "tags": {
+          "description": "Endpoint tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/EndpointPropertiesUpdateParameters"
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "EndpointPropertiesUpdateParameters": {
+      "description": "The JSON object containing endpoint update parameters.",
+      "properties": {
+        "originHostHeader": {
+          "description": "The host header CDN sends along with content requests to origin. The default value is the host name of the origin.",
+          "type": "string"
+        },
+        "originPath": {
+          "description": "A directory path on the origin that CDN can use to retreive content from, e.g. contoso.cloudapp.net/originpath.",
+          "type": "string"
+        },
+        "contentTypesToCompress": {
+          "description": "List of content types on which compression applies. The value should be a valid MIME type.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "isCompressionEnabled": {
+          "description": "Indicates whether content compression is enabled on CDN. Default value is false. If compression is enabled, content will be served as compressed if user requests for a compressed version. Content won't be compressed on CDN when 
```

Comment:
Inconsistent casing!

Reply:
@olydis Are you're referring to the lower camel case in "isCompressionEnabled"?

Reply:
hugh, this commend shows up in the wrong place :( I mean the `x-ms-enum` name `customHttpsProvisioningSubstate`